### PR TITLE
Improve featurization speed for MolGraphConvFeaturizer

### PR DESCRIPTION
**Pull request by nd-02110114**
I fixed the performance when featurizing with `MolGraphConvFeaturizer`.

The previous `MolGraphConvFeaturizer` is slow and it takes 3 minutes to featurize 6000 datapoint.
This indicates it will takes almost 1 hour to featurize QM9 dataset (130000 datapoint). Therefore, I thought I should improve the performance.

The bottleneck is `atom.GetProp('_CIPCode')` for collecting chirality information and `AllChem.ComputeGasteigerCharges(mol)` for collecting partial charge. So, I changed these information to optional ones.

### CODE REVIEW

**Comment by rbharath** [comment]
Looks interesting! Let me know when this is ready for a review :)

File: deepchem/feat/molecule_featurizers/mol_graph_conv_featurizer.py [hunk 1]
```diff
@@ -125,21 +140,32 @@ class MolGraphConvFeaturizer(MolecularFeaturizer):
...
+               use_chirality: bool = False,
+               use_partial_charge: bool = False):
     """
     Parameters
     ----------
-    add_self_edges: bool, default False
-      Whether to add self-connected edges or not. If you want to use DGL,
-      you sometimes need to add explicit self-connected edges.
+    use_edges: bool, default False
+      Whether to use edge features or not.
```
Comment (nd-02110114) [inline deepchem/feat/molecule_featurizers/mol_graph_conv_featurizer.py:151]: I removed `add_self_edges` and added `use_edges ` flag.
This is for simple GCN or GAT models which don't need edge features.

File: deepchem/feat/base_classes.py [hunk 2]
```diff
@@ -267,17 +267,23 @@ def featurize(self, molecules, log_every_n=1000):
...
-      except:
+      except Exception as e:
+        if isinstance(mol, Chem.rdchem.Mol):
+          mol = Chem.MolToSmiles(mol)
         logger.warning(
-            "Failed to featurize datapoint %d. Appending empty array", i)
+            "Failed to featurize datapoint %d, %s. Appending empty array", i,
+            mol)
+        logger.warning("Exception message: {}".format(e))
         features.append(np.array([]))
```
Comment (nd-02110114) [inline deepchem/feat/base_classes.py:287]: I added more logs. I seems that this is noisy for someone...

Comment (nd-02110114) [reply deepchem/feat/base_classes.py:287]: The Tox21 logs

```
Failed to featurize datapoint 95, [I-].[K+]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 255, [Hg+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 659, [Ba+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 985, [TlH2+]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1423, [Cr+3]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1534, [Fe+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1722, [Co+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 1933, [PbH2+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2147, [Fe+3]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2251, [Cu+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2760, [Cd+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 2832, [SnH2+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 4024, [Mn+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 4375, [Be+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 4611, [Zn+2]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
Failed to featurize datapoint 5942, [Br-].[Na+]. Appending empty array
Exception message: zero-size array to reduction operation maximum which has no identity
```

**Comment by nd-02110114** [comment]
@rbharath  This PR is ready to review!

File: deepchem/feat/base_classes.py (hunk 2 above)
Comment (rbharath) [reply deepchem/feat/base_classes.py:287]: I think this is useful! It's clear from this what we're failing on :)

**Review by rbharath — APPROVED** [review]
Looks good! Feel free to merge when ready :)

**Comment by nd-02110114** [comment]
Thank you for review!

**Comment by nd-02110114** [comment]
And, I'm still debugging CGCNNModel ....

**Pull request merged**